server/src/document/document.service.ts

```diff
@@ -151,13 +232,29 @@ export class DocumentService {
         )
       `);
 
+      // We can't determine whether the record id pertains to a Package
+      // or Artifact until we try to query either one.
+      // So if firstPackage is undefined, we then attempt to retrieve an
+      // Artifact
       if (!firstPackage) {
-        const errorMessage = `No document access.`;
-        console.log(`Client attempted to retrieve document ${pathSegment}, but no associated public, submitted packages were found.`);
-        throw new HttpException({
-          "code": "NO_DOCUMENT_ACCESS",
-          "detail": errorMessage,
-        }, HttpStatus.FORBIDDEN);
+        const { records: [firstArtifact] } = await this.crmService.get('dcp_artifactses', `
+          $select=dcp_artifactsid
+          &$filter=dcp_artifactsid eq ${recordId}
+          and (
+            dcp_visibility eq ${ARTIFACT_VISIBILITY.GENERAL_PUBLIC}
+          )
+        `);
+
+        // If the record GUID also returns no Artifact, then there may be
+        // something wrong with the GUID 
+        if (!firstArtifact) {
+          const errorMessage = `No document access.`;
+          console.log(`Client attempted to retrieve document ${pathSegment}, but no associated public, submitted packages or artifacts were found.`);
+          throw new HttpException({
+            "code": "NO_DOCUMENT_ACCESS",
+            "detail": errorMessage,
+          }, HttpStatus.FORBIDDEN);
```

Comment:
Not sure that this should imply a 403. It seems like a 404? In what situation would it be requesting something that isn't found?

Reply:
Very interesting, thanks for looking that stuff up @godfreyyeung